packages/core/src/sync-realtime/index.ts

```diff
@@ -126,14 +129,64 @@ export const createRealtimeSync = (
    * 1 block ahead of the local chain.
    * @returns true if a reorg occurred
    */
-  const handleBlock = async (block: SyncBlock) => {
+  const handleBlock = async ({
+    block,
+    logs,
+    callTraces,
+    transactions,
+    transactionReceipts,
+  }: BlockData) => {
     args.common.logger.debug({
       service: "realtime",
       msg: `Started syncing '${args.network.name}' block ${hexToNumber(block.number)}`,
     });
 
-    const { logs, callTraces, transactions, transactionReceipts } =
-      await extract(block);
+    // Get all addresses from `logs` that are from a factory
+    const childAddresses = new Set<Address>();
+    for (const filter of factories) {
+      if (logs.length > 0) {
+        const _childAddresses = await args.syncStore.filterChildAddresses({
```

Comment:
Might be worth creating an issue to load this once on startup and maintain it in-memory as new blocks come in.